### ISSUE TITLE
SC2015 is still reported if `A && B || C` is used as conditional clause

### ISSUE THREAD

**Issue by jasonkarns** [koalaman/shellcheck#666]
`if A && B || C` is reporting the same warning that is intended for the non-if variant, `A && B || C`.

Actual case:

```sh
if [ -n "$(get_prerelease "${rule#* }")" ] && semver_eq "$(get_number "${rule#* }")" "$(get_number "$ver")" || [ "$rule" = "all" ]; then
```

**Comment by koalaman** [comment]
Well caught. This has been fixed in 1d04754
